core/inbox.go

```diff
@@ -58,31 +59,37 @@ func (ib *Inbox) Pool() *MessagePool {
 // We think that the right model for keeping the message pool up to date is
 // to think about it like a garbage collector.
 func (ib *Inbox) HandleNewHead(ctx context.Context, oldHead, newHead types.TipSet) error {
-	oldBlocks, newBlocks, err := CollectBlocksToCommonAncestor(ctx, ib.chain, oldHead, newHead)
+	oldTips, newTips, err := CollectTipsToCommonAncestor(ctx, ib.chain, oldHead, newHead)
 	if err != nil {
 		return err
 	}
 
 	// Add all message from the old blocks to the message pool, so they can be mined again.
-	for _, blk := range oldBlocks {
-		for _, msg := range blk.Messages {
-			_, err = ib.pool.Add(ctx, msg, uint64(blk.Height))
-			if err != nil {
-				log.Info(err)
+	// The tipsets are iterated in reverse height order, but the order doesn't matter here.
+	for _, tipset := range oldTips {
+		for i := 0; i < tipset.Len(); i++ {
+			block := tipset.At(i)
+			for _, msg := range block.Messages {
+				_, err = ib.pool.Add(ctx, msg, uint64(block.Height))
+				if err != nil {
+					log.Info(err)
+				}
 			}
 		}
 	}
 
 	// Remove all messages in the new blocks from the pool, now mined.
```

Comment:
Shouldn't this comment "Remove all messages in the new blocks" be changed since it's operating on the TipSet now?